webrtc.html

```diff
@@ -11017,74 +10995,76 @@ <h2 id="sec.idp-error-handling">IdP Error Handling</h2>
       <p>Examples assume that <var>sender</var> is an <code><a>RTCRtpSender</a></code>.</p>
       <p>Sending the DTMF signal "1234" with 500 ms duration per tone:</p>
       <pre class="example highlight">
-if (sender.dtmf.canInsertDTMF) {
-    var duration = 500;
-    sender.dtmf.insertDTMF("1234", duration);
-} else
-    log("DTMF function not available");
-</pre>
-      
+        if (sender.dtmf.canInsertDTMF) {
+            const duration = 500;
+            sender.dtmf.insertDTMF("1234", duration);
+        } else {
+            console.log("DTMF function not available");
+        }
+      </pre>
       <p>Send the DTMF signal "123" and abort after sending "2".</p>
       <pre class="example highlight">
-if (sender.dtmf.canInsertDTMF) {
-  sender.dtmf.ontonechange = function (e) {
-      if (e.tone == "2")
-          // empty the buffer to not play any tone after "2"
-          sender.dtmf.insertDTMF("");
-  };
-  sender.dtmf.insertDTMF("123");
-} else
-    log("DTMF function not available");
-</pre>
-      
+        if (sender.dtmf.canInsertDTMF) {
+            sender.dtmf.ontonechange = (e) =&gt; {
+                if (e.tone == "2") {
+                    // empty the buffer to not play any tone after "2"
+                    sender.dtmf.insertDTMF("");
+                }
+            };
+            sender.dtmf.insertDTMF("123");
```

Comment:
I think we should generally encourage methods before onxxx handlers, since we queue a task for all events.

Also, if we wanted to get fancy we could do something like:
```js
if (sender.dtmf.canInsertDTMF) {
    sender.dtmf.insertDTMF("123");
    await new Promise(r => sender.dtmf.ontonechange = e => e.tone == "2" && r());
    // empty the buffer to not play any tone after "2"
    sender.dtmf.insertDTMF("");
} else {
    console.log("DTMF function not available");
}
```

Reply:
> I think we should generally encourage methods before onxxx handlers, since we queue a task for all events.

is it better calling method which fires X event should invoke after onX handler has set ?

Reply:
Either works, is my point. There's a common misunderstanding that you *have to* declare the onxxx first, which is untrue, modulo [bugs](https://bugzilla.mozilla.org/show_bug.cgi?id=1394072). This lets us use `await` as I show.